bin/rule-generator.js

```diff
@@ -0,0 +1,448 @@
+					{
+						id: `${cName}`,
+						evaluate: `${cName}.js`,
+						metadata: {
+							impact: '',
+							messages: {
+								pass: '',
+								fail: '',
+								incomplete: ''
+							}
+						}
+					},
+					undefined,
+					2
+				),
+				dir: outDir
+			};
+			const checkJs = {
+				name: `${cName}.js`,
+				content: `
+				// TODO: Logic for check
+				return true;
+				`,
+				dir: outDir
+			};
+			const checkTestJs = {
+				name: `${cName}.js`,
+				content: `
+				describe('${cName} tests', function() {
+					'use strict';
+					// TODO: Write tests
+				})
+				`,
+				dir: outTestDir
+			};
+			const files = [checkJson, checkJs];
+			if (GET_IS_UNIT_TEST_ASSETS) {
+				files.concat(checkTestJs);
+			}
+			return files;
+		})
+		.reduce((out, item) => {
+			out = out.concat(item);
+			return out;
+		}, []);
+	filesToGenerate = filesToGenerate.concat(checkFiles);
+	return filesToGenerate;
+};
+
+const createFile = fileMeta => {
+	const filePath = `${fileMeta.dir}/${fileMeta.name}`;
+	return new Promise((resolve, reject) => {
+		fs.writeFile(filePath, fileMeta.content, err => {
+			if (err) {
+				reject(err);
+			}
+			resolve(filePath);
+		});
+	});
+};
+
+let axeRulesFileNames;
+let axeRulesIds;
+let axeChecksFileNames;
+let axeChecksIds;
+
+const run = async () => {
+	// ensure axe exists, if not build axe, then start the generator
+	const axeExists = fs.existsSync(axeFile);
+	if (!axeExists) {
+		console.log(
+			chalk.black.bgRed.bold(
+				`Axe does not exist.Triggering build using - 'npm run build'.Rule Generator will continue after build.`
+			)
+		);
+		await cmd.run('npm run build');
+	}
+
+	// compute essentials
+	// 1. Get rule SPEC json file names
+	const ruleJsonSpecFiles = await filehound
+		.create()
+		.paths(DIRECTORIES.RULES)
+		.depth(0)
+		.ext('json')
+		.find();
+	axeRulesFileNames = ruleJsonSpecFiles.map(
+		f =>
+			f
+				.replace('.json', '')
+				.split('/')
+				.reverse()[0]
+	);
+	// 2. Get rule ID's
+	axeRulesIds = ruleJsonSpecFiles.reduce((out, ruleJsonPath) => {
+		const spec = require(ruleJsonPath);
+		out.push(spec.id);
+		return out;
+	}, []);
+	// 3. Get check SPEC json file names
+	const checkJsonSpecFiles = await filehound
+		.create()
+		.paths(DIRECTORIES.CHECKS)
+		.depth(1)
+		.ext('json')
+		.find();
+	axeChecksFileNames = checkJsonSpecFiles.map(
+		f =>
+			f
+				.replace('.json', '')
+				.split('/')
+				.reverse()[0]
+	);
+	// 4. Get all check ID's
+	const axe = require(axeFile);
+	axeChecksIds = ruleJsonSpecFiles.reduce((out, ruleJsonPath) => {
+		const spec = require(ruleJsonPath);
+		const checkIds = []
+			.concat(spec.any || [])
+			.concat(spec.all || [])
+			.concat(spec.none || []);
+		return axe.utils.uniqueArray(out, checkIds);
+	}, []);
+
+	// rule-generator banner
+	showBanner();
+
+	// questions
+	const files = await questionAndGetFilesToCreate();
+
+	// create the files
+	if (!files || !files.length) {
+		console.log(chalk.white.bgRed.bold(`No files to generate.`));
```

Comment:
Same question as above (and below): do these colors work on a variety of themes?

Reply:
Also @marcysutton - I have added some basic docs to the developer guide, would like your opinion on the same.